scene/2d/audio_stream_player_2d.h

```diff
@@ -88,6 +88,9 @@ class AudioStreamPlayer2D : public Node2D {
 	float max_distance = 2000.0;
 	float attenuation = 1.0;
 
+	bool has_playback_lock = false;
+	Mutex playback_lock_mutex;
```

Comment:
The several `playback_lock_mutex` are for making the various player nodes thread-safe in regards to the so-called playback lock, right? Nodes in general are not required to be thread-safe and users are expected to do their own locking. Therefore, maybe those mutexes can go away.

Not super important, though. I'd like to read more opinions on this.

Reply:
I'm a bit more worried about the `play()` call racing with the audio thread's `_mix_audio()` calls, but I guess that two `play()` calls could also race, or potentially a `play()` and a an exit tree notification even.

This code difficult to reason about, and it's not entirely clear how high the risk of a race condition actually is, but it is clear to me that if a race does occur, the audio driver's count of sources that are currently playing could be thrown off by one, which would cause it to never sleep or (worse) not wake up despite there being a source that wants to play audio.

By setting the bool and calling the `notify_source_playing`/`notify_source_stopped_playing` atomically I'm avoiding that risk. I'm also very interested to hear other thoughts though, since my C++ is a bit rusty.